kip-0010.md

```diff
@@ -0,0 +1,198 @@
+   - Existing scripts continue to function as before
+
+### 6. Reserved Opcodes
+The following opcodes are reserved for future expansion:
+- OpTxVersion (0xb2)
+- OpTxLockTime (0xb5)
+- OpTxSubnetId (0xb6)
+- OpTxGas (0xb7)
+- OpTxPayload (0xb8)
+- OpOutpointTxId (0xba)
+- OpOutpointIndex (0xbb)
+- OpTxInputScriptSig (0xbc)
+- OpTxInputSeq (0xbd)
+- OpTxInputBlockDaaScore (0xc0)
+- OpTxInputIsCoinbase (0xc1)
+
+## Rationale
+
+The enhanced opcodes and arithmetic capabilities address several key requirements:
+
+1. Transaction Introspection: Scripts can now examine and validate transaction properties directly, enabling complex conditional logic
+2. Larger Number Support: 8-byte integers allow for precise calculations with large values, essential for financial operations
+3. Future Extensibility: Reserved opcodes provide a clear path for future enhancements
+4. Backward Compatibility: The activation mechanism ensures a smooth network upgrade
+
+These opcodes are designed to work within the existing P2SH framework, maintaining compatibility with current address types while significantly expanding the possibilities for script design.
+
+## Backwards Compatibility
+
+This proposal requires a hard fork, as it introduces new opcodes to the scripting language. Older software will require an update to support these new features. Existing scripts and addresses remain valid, but cannot use the new functionality without being updated.
+
+## Reference Implementation
+
+A reference implementation of the new opcodes and example usage can be found in the following pull request to the rusty-kaspa repository:
+
+[https://github.com/kaspanet/rusty-kaspa/pull/487](https://github.com/kaspanet/rusty-kaspa/pull/487)
+
+Additional examples demonstrating the use of these opcodes can be found in:
+
+[https://github.com/biryukovmaxim/rusty-kaspa/blob/kip-10-mutual-tx/crypto/txscript/examples/kip-10.rs](https://github.com/biryukovmaxim/rusty-kaspa/blob/kip-10-mutual-tx/crypto/txscript/examples/kip-10.rs)
+
+## Example Usage
+
+To illustrate the practical application of the new opcodes, we present three scenarios: a threshold scenario, a shared secret scenario, and a mining pool payment scenario.
+
+### 1. Threshold Scenario
+
+This scenario demonstrates a script that allows for two types of spending conditions:
+
+1. Owner spending: The owner can spend the UTXO by providing a valid signature.
+2. Borrower spending: Anyone can spend the UTXO if they create an output with a value greater than the input by a specified threshold amount, sent to the same script.
+
+### 2. Shared Secret Scenario
+
+This scenario demonstrates a more complex script that allows for:
+
+1. Owner spending: The owner can spend the UTXO by providing a valid signature.
+2. Borrower spending with shared secret: A borrower can spend if they know a shared secret and create an output with a value greater than or equal to the input, sent to the same script.
+
+### 3. Mining Pool Payment Scenario
```

Comment:
Hi, IIUC the point of this use case is to allow small payouts, without suffering from KIP 9 implications. Can you add this to the first paragraph?